File: exercises/tp6/account_test.py
import unittest

from account import (Account, Owner)

moe: Owner = Owner(1, "Moe")
larry: Owner = Owner(2, "Larry")
curly: Owner = Owner(10, "Curly")

class AccountTest(unittest.TestCase):

    def test_creation_owner(self):
        self.assertEqual("Moe", moe.name)

    def test_creation_account(self):
        moe_account = Account(1, moe)
        self.assertEqual(1, moe_account.account_number)
        self.assertEqual(1, moe_account.owner.id_number)
        self.assertEqual("Moe", moe_account.owner.name)
        self.assertEqual(0.0, moe_account.get_balance())

    def test_deposit(self):
        moe_account = Account(1, moe)
        moe_account.deposit(100)
        self.assertEqual(100, moe_account.get_balance())

    def test_withdraw(self):
        moe_account = Account(1, moe)
        moe_account.deposit(100)
        moe_account.withdraw(50)
        self.assertEqual(50, moe_account.get_balance())

    def test_transfer(self):
        moe_account = Account(1, moe)
        larry_account = Account(2, larry)
        moe_account.deposit(100)
        moe_account.transfer(30, larry_account)
        self.assertEqual(70, moe_account.get_balance())
        self.assertEqual(30, larry_account.get_balance())

    def test_list_accounts(self):
        moe_account = Account(1, moe)
        larry_account = Account(2, larry)
        curly_account = Account(3, curly)
        self.assertEqual([moe_account, larry_account, curly_account], Account.get_accounts())

    def test_access_account(self):
        Account(1, moe)
        Account(2, larry)
        Account(3, curly)
        self.assertEqual("Curly", Account.get_account(3).owner.name)

    def test_transactions(self):
        moe_account = Account(1, moe)
        larry_account = Account(2, larry)
        moe_account.deposit(100)
        moe_account.transfer(30, larry_account)
        moe_account.withdraw(40)
        self.assertEqual(30, moe_account.get_balance())
        transactions = moe_account.get_transactions()
        self.assertEquals(3, len(transactions))
        amounts = [t.amount for t in transactions]
        self.assertEquals([100, 30, 40], amounts)

File: exercises/tp10/src/test/java/AccountTest.java
package src.test.java;

import org.junit.jupiter.api.Test;
import src.main.java.Account;
import src.main.java.Owner;
import src.main.java.Transaction;

import java.util.List;
import java.util.stream.Collectors;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class AccountTest {

    Owner moe = new Owner(1, "Moe");
    Owner larry = new Owner(2, "Larry");
    Owner curly = new Owner(10, "Curly");

    @Test
    public void testCreationAccount() {
        Account moeAccount = new Account(1, moe);
        assertEquals(1, moeAccount.getAccountNumber());
        assertEquals(1, moeAccount.getOwner().getIdNumber());
        assertEquals("Moe", moeAccount.getOwner().getName());
        assertEquals(0.0, moeAccount.getBalance());
    }

    @Test
    public void testDeposit() {
        Account moeAccount = new Account(1, moe);
        moeAccount.deposit(100);
        assertEquals(100, moeAccount.getBalance());
    }

    @Test
    public void testWithdraw() {
        Account moeAccount = new Account(1, moe);
        moeAccount.deposit(100);
        moeAccount.withdraw(50);
        assertEquals(50, moeAccount.getBalance());
    }

    @Test
    public void testTransfer() {
        Account moeAccount = new Account(1, moe);
        Account larryAccount = new Account(2, larry);
        moeAccount.deposit(100);
        moeAccount.transfer(30, larryAccount);
        assertEquals(70, moeAccount.getBalance());
        assertEquals(30, larryAccount.getBalance());
    }

    @Test
    public void testListAccounts() {
        Account moeAccount = new Account(1, moe);
        Account larryAccount = new Account(2, larry);
        Account curlyAccount = new Account(3, curly);
        assertEquals(List.of(moeAccount, larryAccount, curlyAccount), Account.getAccounts());
    }

    @Test
    public void testAccessAccount() {
        Account accountMoe = new Account(1, moe);
        Account accountLarry = new Account(2, larry);
        Account accountCurly = new Account(3, curly);
        assertEquals("Curly", Account.getAccount(3).getOwner().getName());
    }

    @Test
    public void testTransactions() {
        Account moeAccount = new Account(1, moe);
        Account larryAccount = new Account(2, larry);
        moeAccount.deposit(100);
        moeAccount.transfer(30, larryAccount);
        moeAccount.withdraw(40);
        assertEquals(30, moeAccount.getBalance());
        List<Transaction> transactions = moeAccount.getTransactions();
        assertEquals(3, transactions.size());
        List<Integer> amounts = transactions.stream().map(Transaction::getAmount).collect(Collectors.toList());
        assertEquals(List.of(100, 30, 40), amounts);
    }
}
